### ISSUE TITLE
Build Pipeline Failure: Cannot read properties of undefined (reading 'Templating') in Highcharts accessibility module

### ISSUE THREAD

**Issue by Hassona97** [highcharts/highcharts#23394]
#### Expected behaviour
The Highcharts accessibility module should initialize properly when imported and called with the main Highcharts object, enabling accessibility features for charts without causing TypeScript compilation errors.

#### Actual behaviour
When importing and initializing the Highcharts accessibility module in a TypeScript React application, the build fails with the error:
```
TypeError: Cannot read properties of undefined (reading 'Templating')
    at /app/node_modules/.pnpm/highcharts@12.3.0/node_modules/highcharts/modules/accessibility.js:12:113
    at Object.<anonymous> (/app/node_modules/.pnpm/highcharts@12.3.0/node_modules/highcharts/modules/accessibility.js:12:1137)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1024:12)
    at cjsLoader (node:internal/modules/esm/translators:348:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:297:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at ModuleLoader.import (node:internal/modules/esm/loader:316:24)
```
#### Live demo with steps to reproduce

```
import 'highcharts/modules/accessibility';

 const baseConfig: Highcharts.Options = {
    accessibility: {
      enabled: true,
```
#### Product version

Core Highcharts Packages
highcharts: 12.3.0
highcharts-react-official: 3.2.2

Build & Runtime Environment
Node.js: >=20.14.0 (from engines)
pnpm: >=10.13.1 / 10.13.1 (package manager)
TypeScript: 5.8.3
Vite: 6.3.5 (build tool)
React & Framework
React: 18.3.1
React DOM: 18.3.1
React Router: 7.5.3
@types/react: 18.3.23
@types/react-dom: 18.3.7

Related Build Tools
@vitejs/plugin-basic-ssl: 2.1.0
vite-plugin-checker: 0.9.3
vite-tsconfig-paths: 5.1.4
@swc/core: 1.13.2

**Comment by bre1470** [comment]
Hello, @Hassona97 !

Thank you for your report.
Please provide more code, so we can get a better understanding of your issue.
It looks like you do not import Highcharts itself or assign it to our react component.

Besides I suggest to take your Highcharts imports from our ESM bundles as these are known to have improved compatibility with modern frameworks.

In the provided code snippet you would replace the statement `import 'highcharts/modules/accessibility';` with `import 'highcharts/esm/modules/accessibility.js';`.
The same applies to Highcharts which should be imported with the following statement: `import 'highcharts/esm/highcharts.js';`.

Please let us know if the ESM bundles solve the issue.
Otherwise we will need more code to be able to help you.

**Comment by Hassona97** [comment]
Thanks @bre1470 for your response. Trying this import `highcharts/esm/modules/accessibility.js` instead fixed the error I was getting. But there is another one that came up.

<img width="792" height="415" alt="Image" src="https://github.com/user-attachments/assets/742e4ed9-90f0-4d88-b3aa-e9602732ba46" />

I tried to use another import when I dig deep in the node_modules ` highcharts/es-modules/Accessibility/Accessibility` this one and still get a similar error 

<img width="833" height="349" alt="Image" src="https://github.com/user-attachments/assets/66d066b9-dbb3-4e56-a366-278e46b99244" />

My package.json is contaiing the `type:module` which is suggested in the SCs attached. Please let me know what part of code I need to provide for you.

**Comment by bre1470** [comment]
Hello, @Hassona97 !

Thank you for the shared details.

* With `module` mode in your `package.json` you should make sure to solely load the bundles in `highcharts/esm/`.

* Additionally you might need to adjust your `tsconfig.json` accordingly. Something like the following settings should work:
  ```json
  {
    "compilerOptions": {
      "module": "ES2020",
      "moduleResolution": "Node",
  ```

* In ESM / module mode you also need to make sure to use `.js` file extension for all `highcharts/esm/` paths.

* You can also take a look at this [example project](https://github.com/bre1470/highcharts-react-ts-vite) to compare it to your project configuration and make adjustments.

* To help you further we need the `compilerOptions` in your `tsconfig.app.json` and your code of the Highcharts react component, especially imports and `<HighchartsReact />` statement with assigned attributes. This should follow our recommended examples as highlighted in the [package README](https://www.npmjs.com/package/highcharts-react-official).

Regarding your second screenshot: The `highcharts/es-modules/` files are a synchronous solution for bundlers like Webpack, where only `highcharts/es-modules/masters/` can be imported. It is not suitable for recent versions of Vite.js.

Please let us know about the progress.

**Comment by Hassona97** [comment]
Hello @bre1470,

1- What do you mean by solely load the bundles in `highcharts/esm/`.

2- my tsconfig is using 
```
 "module": "ESNext",
 "moduleResolution": "Bundler",

```
However, I changed to what you send but my application failed to build.

3- I am using .js extensions

Here is my compilerOptions tsconfig.json 
```
  "compilerOptions": {
    "allowJs": true,
    "alwaysStrict": true,
    "baseUrl": ".",
    "declaration": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "incremental": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "lib": ["DOM", "DOM.Iterable", "ES2022"],
    "module": "ESNext",
    "moduleResolution": "Bundler",
    "noFallthroughCasesInSwitch": false,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUncheckedIndexedAccess": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": false,
    "target": "ES2022",
    "tsBuildInfoFile": "./.cache/.tsBuildInfo/tsconfig.tsbuildinfo",
    "types": ["@react-router/node", "vite/client"],
    "paths": {
      "@shared/*": ["./app/*"],
      "~/*": ["./app/*"]
    },
    "noEmit": true
  }
```

and here is my use for the highcharts

```
import Highcharts from 'highcharts/esm/highcharts.js';
import HighchartsReact from 'highcharts-react-official';
import 'highcharts/esm/modules/accessibility.js';

const baseConfig: Highcharts.Options = {
      accessibility: {
        enabled: true,
      }
}

<HighchartsReact
            highcharts={Highcharts}
            id={chartId}
            options={mergedOptions}
          />
```

please let me know If I want to share anything else.

**Comment by bre1470** [comment]
Hello @Hassona97 !

Thank you for the details.
I was able to produce a similar issue.

The issue is caused by the way the components are bundled.
In the React Router setup the components by default load without a web browser to improve performance.
Highcharts on the other side needs access to the web browser for rendering purposes.

**Solution:** One has to add `'use client';` to project files, where Highcharts is imported, so that the component is loaded in the traditional way.

I created an example based on your details: https://github.com/bre1470/highcharts-issue-23394/blob/main/app/chart/chart.tsx

Please let us know if this solves your issue as well.

**Comment by bre1470** [comment]
An update:

It looks like `'use client';` is not completely solving the rendering issue and it was necessary to turn off server-side rendering in `react-router.config.ts`:

```TypeScript
import type { Config } from "@react-router/dev/config";

export default {
  // Config options...
  // Server-side render by default, to enable SPA mode set this to `false`
  ssr: false, // <-- fixes issue
} satisfies Config;
```
https://github.com/bre1470/highcharts-issue-23394/blob/main/react-router.config.ts

**Comment by Hassona97** [comment]
Thanks @bre1470  for your help, But we need the `ssr` to be set to true in our application for other packages. Is there any other possible solution ?

**Comment by bre1470** [comment]
Hello, @Hassona97!

Thank you for the information.
We will discuss the issue internally and work on a solution.

**Comment by goransle** [comment]
> Thanks [@bre1470](https://github.com/bre1470) for your help, But we need the `ssr` to be set to true in our application for other packages. Is there any other possible solution ?

Hi @Hassona97! If you are using React-Router with the newer framework mode, the way to force a route to be client-side rendered is to set the [`clientLoader` option](https://reactrouter.com/start/framework/route-module#clientloader):

```jsx
import { Chart } from "../chart/chart";

export async function clientLoader() {
    return true;
}

export default function Test() {
    return <Chart />;
}
```

Hope this helps, but if this is not the case, are you able to share more about your setup? A recreation on i.e. Stackblitz would be particularly useful.

**Comment by Hassona97** [comment]
Hello @goransle, Unfortunately this also didn't work. I think recreation with the same setup would be hard since it fails in the production build on building the docker image, otherwise it's working fine on local development. What else in the setup would help you that I can provide.

**Comment by goransle** [comment]
@Hassona97 I see. Are you able to share the build command that runs in docker?

**Comment by Hassona97** [comment]
```
RUN pnpm install --frozen-lockfile
RUN pnpm build ("build": "dotenvx run -- react-router build")
```
@goransle this what you need ?

**Comment by goransle** [comment]
@Hassona97 Unfortunately I'm still not able to reproduce this issue on my end. If you are able to share a minimal recreation that would be very much appreciated.

Alternatively, [our new React wrapper](https://www.npmjs.com/package/@highcharts/react), currently in beta, could perhaps help as it handles imports internally.

**Comment by goransle** [comment]
@Hassona97 Closing this for now. If you can come up with a reproduction please feel free to reopen it!
